### PR TITLE
Added gradle task for versioning and edited workflows

### PR DESCRIPTION
**Pull request by krossgg**
## What
No more grep

## Implementation Details
Runs new gradle task `printVersion` whenever it needs the version number in the form `{mc-ver}-{mod-ver}`, e.g. `1.20.1-1.4.2`

### CODE REVIEW

File: .github/workflows/auto-build.yml [hunk 1]
```diff
@@ -35,10 +35,10 @@ jobs:
       - name: Get Version
         id: var
         run: |
-          MESSAGE=$(ls build/libs/* | grep dev.jar -v | grep sources.jar -v | grep slim.jar -v | awk -F 'gtceu-|-SNAPSHOT.jar' '{print $2}')
-          mv "build/libs/gtceu-$MESSAGE-SNAPSHOT.jar" "build/libs/gtceu-$MESSAGE-build_${{ github.run_number }}-SNAPSHOT.jar"
-          mv "build/libs/gtceu-$MESSAGE-SNAPSHOT-slim.jar" "build/libs/gtceu-$MESSAGE-build_${{ github.run_number }}-SNAPSHOT-slim.jar"
-          echo "version=$MESSAGE-build_${{ github.run_number }}" >> $GITHUB_OUTPUT
+          VER=$(./gradlew -q printVersion)
+          BUILD_VER=$VER-build_${{ github.run_number }}
+          for jar in ./build/libs/*; do mv "$jar" "${jar/${VER}-SNAPSHOT/${BUILD_VER}-SNAPSHOT}";done 2>/dev/null
```
Comment (screret) [inline .github/workflows/auto-build.yml:40]: doesn't this move it to a different folder?

Comment (krossgg) [reply .github/workflows/auto-build.yml:40]: No, the `for in` syntax sets `$jar` to the entire filepath of the files it finds, so mv and the substr replacement use the full path

Comment (screret) [reply .github/workflows/auto-build.yml:40]: no I meant the `jar/${VER}-SNAPSHOT/${BUILD_VER}-SNAPSHOT` part

**Review by screret — APPROVED** [review]
gotta wait for modmaven to get back up so we can merge

**Pull request merged**